### PR TITLE
Nonholonomic slotcar turning fixes, add turning multiplier offset parameter, rename compute_ds variables

### PR DESCRIPTION
**Pull request by ddengster**
### Fixed bugs

- The nonholonomic slotcar plugin had a problem with it's turning due to the computation of the turn delta, this PR fixes it via using the arcosine of the dot product.

- Another problem is the decceleration applied to velocities in `compute_ds`, making our slotcar enter turns from `0` velocity and in turn making the turn look weird. To fix this we added `target_velocity` as a targeted final velocity as `target_s` reaches `0`.

- Added  `_turning_right_angle_mul_offset` as an added multiplier applied to the turning radius for right-angled turns within 20 degrees. Eg. A value of `0.5` for `_turning_right_angle_mul_offset` makes for a multiplier `1.5` when is applied to a `90 degree` turning radius, and linearly downscales to `1.0` for a `70 degree` turning radius. This compensates for turning acc/deccelerations making turns 'overshoot'.

- Corrects up some variables named `velocities` when they were in fact `displacements`.

- Removes unneeded variables in various spots

In conjunction with https://github.com/open-rmf/rmf_demos/pull/82

### CODE REVIEW

File: rmf_robot_sim_common/include/rmf_robot_sim_common/utils.hpp [hunk 1]
```diff
@@ -60,7 +60,8 @@ double compute_ds(
   const double v_max,
   const double accel_nom,
   const double accel_max,
-  const double dt);
+  const double dt,
+  const double v_target = 0.0);
```
Comment (luca-della-vedova) [inline rmf_robot_sim_common/include/rmf_robot_sim_common/utils.hpp:64]: Nit, there was some discussion here about the `compute_ds` and `compute_desired_rate_of_change` being basically the same function, can we take the chance to remove one of the two?

Comment (ddengster) [reply rmf_robot_sim_common/include/rmf_robot_sim_common/utils.hpp:64]: There's a line that differs between functions by double the divisor at https://github.com/open-rmf/rmf_simulation/pull/41/files/d55cfacfe9310deef0b6012e1a81ab2c7bb51394#diff-3bd4615902d61050ebeb9f584b212fc5d5ef07c1eed12cb5fa7814dbed7d59a4R41
but you're right let me remove it and test it with the other robots

**Review by luca-della-vedova — COMMENTED** [review]
Works great! Just a minor comment.
I guess for the future we might want to have a different parameter for the velocity in bends to be different (smaller) from the velocity in a straight path, but we can address that in the future.

File: rmf_robot_sim_common/include/rmf_robot_sim_common/utils.hpp (hunk 1 above)
Comment (ddengster) [reply rmf_robot_sim_common/include/rmf_robot_sim_common/utils.hpp:64]: delayed, will have to tweak the turning_right_angle_mul_offset values again

**Review by luca-della-vedova — APPROVED** [review]

**Pull request merged**